magicblock-api/src/magic_validator.rs

```diff
@@ -691,6 +690,15 @@ impl MagicValidator {
     pub fn ledger(&self) -> &Ledger {
         &self.ledger
     }
+
+    /// Prepares RocksDB for shutdown by cancelling all Manual compactions
+    /// This speeds up `stop` as it doesn't have to await for compaction cancellation
+    /// Calling this still allows to write or read from DB
+    pub fn prepare_ledger_for_shutdown(&mut self) {
+        self.ledger_truncator.stop();
+        // Calls & awaits until manual compaction is canceled
+        self.ledger.cancel_manual_compactions();
```

Comment:
Will this call block until everything is canceled?

Reply:
Yeap, based on source code